common-arcana.lic

```diff
@@ -603,26 +603,30 @@ module DRCA
       cast_spell({ 'abbrev' => 'cv', 'mana' => 1, 'prep_time' => 5 }, settings)
     end
 
-    release_cyclics if data['cyclic']
-    DRC.bput('release spell', 'You let your concentration lapse', "You aren't preparing a spell") unless checkprep == 'None'
-    DRC.bput('release mana', 'You release all', "You aren't harnessing any mana")
-
     if data['ritual']
```

Comment:
I've not cast ritual spells yet, but is there any negative effect by not releasing spell/mana before attempting the ritual here?

Reply:
This is not new/being edited/or involved with this edit.  It's just showing red because it's been rearranged.
 And yes.  You always release everything before hand so you don't have too much mana or a previously prepped spell.

Reply:
Sorry, what I meant is if you were to cast a ritual spell then old lines 607 and 608 do `release spell` and `release mana` before `ritual(data, settings)` is called, which try to prepare/cast the spell. It has lost the pre-step of releasing spell/mana before attempting to cast the ritual if I'm reading the ritual method correctly.

```ruby
  def ritual(spell, settings)
    DRC.retreat(settings.ignored_npcs) unless spell['skip_retreat']
    DRC.release_invisibility
    DRC.set_stance('shield')

    command = 'prepare'
    command = spell['prep'] if spell['prep']
    command = spell['prep_type'] if spell['prep_type']

    return unless prepare?(spell['abbrev'], spell['mana'], spell['symbiosis'], command, spell['tattoo_tm'], spell['runestone_name'], spell['runestone_tm'])
    find_focus(spell['focus'], spell['worn_focus'], spell['tied_focus'], spell['sheathed_focus'])

    invoke(spell['focus'], nil, nil)
    stow_focus(spell['focus'], spell['worn_focus'], spell['tied_focus'], spell['sheathed_focus'])
    DRC.retreat(settings.ignored_npcs) unless spell['skip_retreat']

    waitcastrt?
    return unless cast?(spell['cast'], spell['symbiosis'], spell['before'], spell['after'])
    DRC.retreat(settings.ignored_npcs) unless spell['skip_retreat']
  end
  ```